src/user/library/index.ts

```diff
@@ -148,75 +124,98 @@ export function editCampaignValues(
 
   return {
+    id: campaign.id,
     adSets: campaign.adSets.map((adSet) => {
       const seg = adSet.segments ?? ([] as Segment[]);
 
       return {
-        ...adSet,
         id: adSet.id,
-        conversions: adSet.conversions ?? [],
-        oses: adSet.oses ?? ([] as OS[]),
-        segments: adSet.segments ?? ([] as Segment[]),
+        conversions: (adSet.conversions ?? []).map((c) => ({
+          id: c.id,
+          type: c.type,
+          observationWindow: c.observationWindow,
+          urlPattern: c.urlPattern,
+        })),
+        oses: (adSet.oses ?? []).map((o) => ({ name: o.name, code: o.code })),
+        segments: (adSet.segments ?? []).map((o) => ({
+          name: o.name,
+          code: o.code,
+        })),
         isNotTargeting: seg.length === 1 && seg[0].code === "Svp7l-zGN",
         name: adSet.name || adSet.id.split("-")[0],
-        creatives: creativeList(adSet.ads),
-      };
+        creatives: creativeList(advertiserId, adSet.ads),
+      } as AdSetForm;
     }),
+    isCreating: false,
     advertiserId,
-    hasPaymentIntent: campaign.hasPaymentIntent ?? false,
-    creatives: creativeList(ads).map((a) => a.id!),
+    creatives: creativeList(advertiserId, ads),
     newCreative: initialCreative,
-    isCreating: false,
+    currency: campaign.currency,
     price: price.toNumber(),
     billingType: billingType,
     validateStart: false,
     budget: campaign.budget,
-    currency: campaign.currency,
     dailyBudget: campaign.dailyBudget,
-    dailyCap: campaign.dailyCap,
     endAt: campaign.endAt,
     format: campaign.format,
-    geoTargets: campaign.geoTargets ?? ([] as GeocodeInput[]),
+    geoTargets: (campaign.geoTargets ?? []).map((g) => ({
+      code: g.code,
+      name: g.name,
+    })),
     name: campaign.name,
-    pacingStrategy: campaign.pacingStrategy,
     startAt: campaign.startAt,
     state: campaign.state,
     type: "paid",
-    stripePaymentId: campaign.stripePaymentId,
     paymentType: campaign.paymentType,
   };
 }
 
-function creativeList(ads?: AdFragment[] | null): Creative[] {
+function creativeList(
+  advertiserId: string,
+  ads?: AdFragment[] | null,
+): Creative[] {
   return _.uniqBy(
     (ads ?? [])
-      .filter((ad) => ad.creative != null && ad.state !== "deleted")
+      .filter((ad) => ad.creative !== null && ad.state !== "deleted")
       .map((ad) => {
         const c = ad.creative;
         return {
+          ...transformCreativeFragment(c, advertiserId),
           creativeInstanceId: ad.id,
-          id: c.id,
-          name: c.name,
-          targetUrl: c.payloadNotification!.targetUrl,
-          title: c.payloadNotification!.title,
-          body: c.payloadNotification!.body,
-          targetUrlValidationResult: "",
-          state: c.state,
         };
       }),
     "id",
   );
 }
 
+export function transformCreativeFragment(
+  c: CreativeFragment | Creative,
+  advertiserId: string,
+) {
```

Comment:
nit: could you add a return type or change the function name so I know what this is transforming the creativeFragment _into_?